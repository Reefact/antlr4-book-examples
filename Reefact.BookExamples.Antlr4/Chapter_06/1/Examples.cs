#region Usings declarations

using ApprovalTests;
using ApprovalTests.Reporters;

using Xunit;

#endregion

namespace Reefact.BookExamples.Antlr4.Chapter_06._1 {

    [UseReporter(typeof(VisualStudioReporter))]
    public class Examples {

        [Fact]
        public void Tree() {
            // Setup
            GRun grun = GRun.ReadResource("data.csv", 6, 1);
            // Exercise
            string listStyleTree = grun.ToLispStyleTree();
            // Verify
            Approvals.Verify(listStyleTree);
        }

        [Fact]
        public void Tokens() {
            // Setup
            GRun grun = GRun.ReadResource("data.csv", 6, 1);
            // Exercise
            string tokensString = grun.ToTokensString();
            // Verify
            Approvals.Verify(tokensString);
        }

        [Fact]
        public void MermaidTree() {
            // Setup
            GRun grun = GRun.ReadResource("data.csv", 6, 1);
            // Exercise
            string mermaidStyleTree = grun.ToMermaidStyleTree();
            // Verify
            Approvals.Verify(mermaidStyleTree);
        }

    }

}