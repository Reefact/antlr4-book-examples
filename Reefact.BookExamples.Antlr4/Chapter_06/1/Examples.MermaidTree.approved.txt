graph TD
	1["file"] --> 2["hdr"]
	2 --> 3["row"]
	3 --> 4["field"]
	4 --> 5["Details"]
	3 --> 6[","]
	3 --> 7["field"]
	7 --> 8["Month"]
	3 --> 9[","]
	3 --> 10["field"]
	10 --> 11["Amount"]
	3 --> 12["#0092;r"]
	3 --> 13["#0092;n"]
	1 --> 14["row"]
	14 --> 15["field"]
	15 --> 16["Mid Bonus"]
	14 --> 17[","]
	14 --> 18["field"]
	18 --> 19["June"]
	14 --> 20[","]
	14 --> 21["field"]
	21 --> 22["#0034;$2,000#0034;"]
	14 --> 23["#0092;r"]
	14 --> 24["#0092;n"]
	1 --> 25["row"]
	25 --> 26["field"]
	25 --> 27[","]
	25 --> 28["field"]
	28 --> 29["January"]
	25 --> 30[","]
	25 --> 31["field"]
	31 --> 32["#0034;#0034;#0034;zippo#0034;#0034;#0034;"]
	25 --> 33["#0092;r"]
	25 --> 34["#0092;n"]
	1 --> 35["row"]
	35 --> 36["field"]
	36 --> 37["Total Bonuses"]
	35 --> 38[","]
	35 --> 39["field"]
	39 --> 40["#0034;#0034;"]
	35 --> 41[","]
	35 --> 42["field"]
	42 --> 43["#0034;$5,000#0034;"]
	35 --> 44["#0092;r"]
	35 --> 45["#0092;n"]
